ndb_api/snapshot.go

```diff
@@ -0,0 +1,94 @@
+package ndb_api
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Fetches all the snapshots on the NDB instance and returns a slice of the snapshots
+func GetAllSnapshots(ctx context.Context, ndbClient *ndb_client.NDBClient) (snapshots []SnapshotResponse, err error) {
```

Comment:
This API was recently updated to support pagination. It will only return 50 snapshots in a single call. Based on the offset in the response, you might want to fetch inside a while loop - if you need all the snapshots.

Reply:
Discussed with @akshayNu @shenoypritika  don't use /snapshots endpoint.